code/modules/mob/living/carbon/carbon.dm

```diff
@@ -90,6 +90,37 @@
 	if (!M.can_use_hand())
 		return
 
+	if(M.a_intent != I_HELP)
+		var/action
+		switch(a_intent)
+			if(I_GRAB)
+				action = "grabbed"
+			if(I_DISARM)
+				action = "pushed"
+			if(I_HURT)
+				action = "punched"
+		var/t_him = "it"
+		if (src.gender == MALE)
+			t_him = "him"
+		else if (src.gender == FEMALE)
+			t_him = "her"
+		var/show_ssd
+		var/mob/living/carbon/human/H = src
+		if(istype(H)) show_ssd = H.species.show_ssd

```

Comment:
`istype(src)` is generally not a valid construct. Is there truly no better way to do this?